PREREQUISITE.md

```diff
@@ -185,77 +185,67 @@ Let's look up the code.
     private Button button;
+    private ActivityResultLauncher<String[]> multiplePermissionLauncher;
+    final String[] permissions = {
+            Manifest.permission.RECORD_AUDIO
+            // add other permissions
+    };
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(), isGranted -> {
+            if(isGranted.toString().contains("false")) {
+                Log.d("Permissions", "result: " + isGranted.toString());
+                finish();
+            }
+        });
+
         button = findViewById(R.id.button2);
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                int permission = ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO);
-
-                if (permission == PackageManager.PERMISSION_DENIED) {
-                    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                        requestPermissions(new String[]{
-                                Manifest.permission.RECORD_AUDIO}, 1000);
-                    }
+                if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    multiplePermissionLauncher.launch(permissions);
                 }
             }
         });
     }
-```
 
-```java
-   //after requestPermission
-    @Override
-    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-        if(requestCode == 1000) {
-            boolean check_result = true;
-            for (int result : grantResults) {
-                if (result != PackageManager.PERMISSION_GRANTED) {
-                    check_result = false;
-                    break;
-                }
-            }
-            if(check_result == true) {
-
-            } else {
-                finish();
-            }
-        }
-    }
 }
 ```
 
 This is `MainActivity` for adding permission.
 
 ```java
-int permission = ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO);
-
-if (permission == PackageManager.PERMISSION_DENIED) {
-    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-        requestPermissions(new String[]{
-                Manifest.permission.RECORD_AUDIO}, 1000);
+ActivityResultLauncher<String[]> multiplePermissionLauncher = registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(), isGranted -> {
+    if(isGranted.toString().contains("false")) {
+        Log.d("Permissions", "result: " + isGranted.toString());
+        finish();
     }
+});
+
+if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+    multiplePermissionLauncher.launch(permissions);
 }
```

Comment:
```suggestion
String[] permissions = {
    Manifest.permission.RECORD_AUDIO
    // add other permissions
};

// after requestPermissions
ActivityResultLauncher<String[]> multiplePermissionLauncher = registerForActivityResult(new 
ActivityResultContracts.RequestMultiplePermissions(), grantResults -> {
    for(Boolean result : grantResults.values()) {
        if (!result) {
            finish();
        }
    }
});
if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
    multiplePermissionLauncher.launch(permissions);
}
```

Reply:
LGTM !

I understand where I was making a mistake, It seems to be more readable than the example I wrote. 
I really appreciate your help ! 😄